### PR TITLE
Add custom feature converter

### PR DESCRIPTION
**Pull request by piresramon**

### CODE REVIEW

File: t5x/custom.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+             "decoder_input_tokens": decoder_input_tokens,
+             "decoder_loss_weights": seqio.non_padding_position(decoder_target_tokens)}
+
+        return d
+
+    def _convert_features(
+      self, ds: tf.data.Dataset,
+      task_feature_lengths: Mapping[str, int]) -> tf.data.Dataset:
+        """Convert the dataset to be fed to a language model."""
+        # ds = self._pack_or_pad(ds, task_feature_lengths)
```
Comment (rodrigonogueira4) [inline t5x/custom.py:39]: remove?

**Review by rodrigonogueira4 — APPROVED** [review]
Thanks for adding this!

**Comment by rodrigonogueira4** [comment]
We should use `TASK_FEATURE_LENGTHS = {"targets": 2048}`, right?

File: t5x/custom.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    def _convert_example(
+      self, features: Mapping[str, tf.Tensor]) -> Mapping[str, tf.Tensor]:
+        """Convert an LM example into an example with model features."""
+
+        decoder_target_tokens = features["targets"][1:]
+        decoder_input_tokens = features["targets"][:-1]
+
+        d = {"decoder_target_tokens": decoder_target_tokens,
+             "decoder_input_tokens": decoder_input_tokens,
+             "decoder_loss_weights": seqio.non_padding_position(decoder_target_tokens)}
```
Comment (rodrigonogueira4) [inline t5x/custom.py:31]: In this function, the PAD token id is zero (https://github.com/google/seqio/blob/a255e24aebc16e0c75cf27638e9f3b4b8fdc73b5/seqio/feature_converters.py#L156). But in GPTimbau's tokenizer, the EOS id is also zero, right? Thus, will EOS tokens have no loss?

File: t5x/custom.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Copyright 2022 The T5X Authors.
```
Comment (ZanezZephyrs) [inline t5x/custom.py:1]: do we need this license comment in the top?

**Review by ZanezZephyrs — APPROVED** [review]

File: t5x/custom.py (hunk 3 above)
Comment (hugoabonizio) [reply t5x/custom.py:1]: I think this preamble can be removed for now

**Review by hugoabonizio — APPROVED** [review]

**Pull request merged**